src/workflow/Workflow.js

```diff
@@ -0,0 +1,368 @@
+const Joi = require('joi')
```

Comment:
The methods of this class are oriented towards a functional approach, which means any update to the workflow returns a new workflow

Reply:
It makes everything more verbose, but I don't want to go down the road with class instances - I think they are more confusing to use in our mainly functional code on the frontend and backend.